### DISCUSSION TITLE
Figure out which modules should have citations

### DISCUSSION THREAD

**Discussion by fefossa** [CellProfiler/CellProfiler#4776] [General]
To implement the citation generator for CellProfiler in a way that has citations for not only the cellprofiler-plugins modules but also the main modules of CellProfiler, we were talking about defining which of the modules will be updated to have a doi dictionary.

So the way the code works now is to get the references from a dictionary that lives inside each module ([RunCellpose](https://github.com/CellProfiler/CellProfiler-plugins/blob/9d91b04fc0a4e2ff4af5c0b773c9582e8cb014ac/runcellpose.py#L76) example).

I pull the modules that have references now to see if we want to have them included in the citation generator list. I'm listing them into **(1) add doi or (2) don't add doi functionality** (based on my judgment, which is only by taking a look at the papers), so please suggest what the final list should be!

**(1) Add doi**

- [ ] 1 - [IdentifyDeadWorms](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/identifydeadworms.py#L38), [StraightenWorms](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/straightenworms.py#L76), and [UntangleWorms](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/untangleworms.py#L86): it's a published toolbox
- [ ] [RunImageJMacro](https://github.com/CellProfiler/CellProfiler/blob/master/cellprofiler/modules/runimagejmacro.py) doesn't have any reference/citation

**(2) Don't add doi**
- [ ] 2 - [MeasureTexture:](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/measuretexture.py#L140) general citation, about texture features
- [ ]  5 - [MakeProjection:](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/makeprojection.py#L154) related to a specific type of projection (Variance)
- [ ] 3 - [TrackObjects:](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/trackobjects.py#L490) there are two citations, one specific for LAP method and the other one is not specific cited during the text (both are from same author)
- [ ] 4 - [MeasureImageQuality:](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/measureimagequality.py#L143) so there are a few citations in this one.
 Specific: [Haralick correlation](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/measureimagequality.py#L1237)
For the whole module: [Bray 2012](https://journals.sagepub.com/doi/full/10.1177/1087057111420292) 
- [ ] 6 - [MeasureImageOverlap](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/measureimageoverlap.py#L87) and [MeasureObjectOverlap:](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/measureobjectoverlap.py#L55) only one of the citations is mentioned to be used [inside one of the functions](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/measureobjectoverlap.py#L576) made for this module
- [ ] 7 - [IdentifyPrimaryObjects:](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/identifyprimaryobjects.py#L205) the five citations are being used to reference the declumping strategy. Seems a lot to cite when using ID1
- [ ] 8 - [MeasureColocalization:](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/measurecolocalization.py#L73) just citing a review that can be helpful to understand colocalization
- [ ] 9 - [CalculateStatistics:](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/calculatestatistics.py#L609) cites an adaptation of a code from Ravkin.
- [ ] 10 - [Threshold:](https://github.com/CellProfiler/CellProfiler/blob/09370bd58cd7592c97bf8419ceab40ef3932407b/cellprofiler/modules/threshold.py#L372) the reference seems to be related to the threshold methods (not specific)

**Comment by bethac07** [comment]
Thanks for pulling this together!

I think we do want TrackObjects, though we can mention it's for the LAP tracker; I don't think we'd need to do RunImageJMacro (it would just be the CP4 paper, in any way). I'd need to look at the MIQ papers (especially Bray 2012) to decide on that one.
